src/main/java/org/opentripplanner/transit/model/site/FlexLocationGroup.java

```diff
@@ -11,37 +12,40 @@
 import org.opentripplanner.transit.model.basic.I18NString;
 import org.opentripplanner.transit.model.basic.WgsCoordinate;
 import org.opentripplanner.transit.model.framework.FeedScopedId;
-import org.opentripplanner.transit.model.framework.TransitEntity;
-import org.opentripplanner.transit.model.site.Stop;
-import org.opentripplanner.transit.model.site.StopLocation;
+import org.opentripplanner.transit.model.framework.TransitEntity2;
 
 /**
  * A group of stopLocations, which can share a common Stoptime
  */
-public class FlexLocationGroup extends TransitEntity implements StopLocation {
+public class FlexLocationGroup
+  extends TransitEntity2<FlexLocationGroup, FlexLocationGroupBuilder>
+  implements StopLocation {
 
   private final Set<StopLocation> stopLocations = new HashSet<>();
-  private I18NString name;
+  private final I18NString name;
   private GeometryCollection geometry = new GeometryCollection(
```

Comment:
Should this be final and the `stopLocations` be an immutable set, as the entities are all available when creating a `FlexLocationGroup`, and could be pushed to the builder